src/exports.cc

```diff
@@ -925,5 +932,27 @@ Word get_log_level(Word result_level_uint32_ptr) {
   return WasmResult::Ok;
 }
 
+Word wasi_unstable_fd_fdstat_set_flags(Word /*fd*/, Word /*flags*/) {
+  // Don't support reading of any files.
+  return 52; // __WASI_ERRNO_ENOSYS
+}
+
+Word wasi_unstable_fd_filestat_get(Word /*fd*/, Word /*statOut*/) {
+  // Don't support reading of any files.
+  return 52; // __WASI_ERRNO_ENOSYS
+}
+
+Word wasi_unstable_fd_readdir(Word /*fd*/, Word /*buf*/, Word /*buf_len*/, int64_t /*cookie*/,
+                              Word /*bufused*/) {
+  // Don't support reading of any files.
+  return 52; // __WASI_ERRNO_ENOSYS
+}
+
+Word wasi_unstable_path_filestat_get(Word /*fd*/, Word /*flags*/, Word /*path*/, Word /*path_len*/,
+                                     Word /*statOut*/) {
+  // Don't support reading of any files.
+  return 52; // __WASI_ERRNO_ENOSYS
+}
+
```

Comment:
FYI, we're using a mix of `44` (`ENOENT`), `8` (`EBADF`) and `52` (`ENOSYS`) in the existing filesystem-related hostcalls, so it would be good to know why did we select `ENOSYS` here.

To be honest, I'm not sure why are we returning `EBADF` in `wasi_unstable_fd_prestat_get` and `ENOSYS` in `wasi_unstable_fd_prestat_dir_name`, but I imagine it was driven by supported error code paths in one of the languages.

Reply:
I took a look at the wasm module, and it turned out some of these were unnecessary. This PR was drafted very early in the process of getting the Go SDK working, and I cannot recall the original purpose of these hostcalls.

They are now removed. My apologies.